### PR TITLE
Allow optionals in Array sectionBy function

### PR DESCRIPTION
**Pull request by skytan4**

### CODE REVIEW

File: Swiftification/Array.swift [hunk 1]
```diff
@@ -99,10 +99,10 @@ public extension Array {
     }
     
     /// Sections the ordered array by the string specified in the closure, preserving the order of the original array.
-    func sectionBy(@noescape sectionFunction: (Element) -> String) -> [(title: String, items: [Element])] {
-        var lastTitle = ""
+    func sectionBy(@noescape sectionFunction: (Element) -> String?) -> [(title: String?, items: [Element])] {
```
Comment (hiltonc) [inline Swiftification/Array.swift]: Can you just change this to allow any element? Like:

    func sectionBy<T>(@noescape sectionFunction: (Element) -> T) -> [(T, [Element])]

Comment (skytan4) [inline Swiftification/Array.swift]: No problem!

File: Swiftification/Array.swift [hunk 2]
```diff
@@ -99,26 +99,26 @@ public extension Array {
     }
     
     /// Sections the ordered array by the string specified in the closure, preserving the order of the original array.
-    func sectionBy(@noescape sectionFunction: (Element) -> String) -> [(title: String, items: [Element])] {
-        var lastTitle = ""
+    func sectionBy<T: Equatable>(@noescape sectionFunction: (Element) -> T) -> [(T, [Element])] {
+        var lastTitle: T? = nil
```
Comment (hiltonc) [inline Swiftification/Array.swift]: The function comment, `lastTitle`, and `title` need an update. Maybe we can refer to it as the section header?

Also, this is @nickmshelley's baby, so looping him in.

Comment (skytan4) [inline Swiftification/Array.swift]: How does `sectionHeader` and `lastSectionHeader` sound to everyone?

Comment (nickmshelley) [inline Swiftification/Array.swift]: `Name` implies `String`, but `Header` doesn't, so I like the `Header` variants better.

**Comment by hiltonc** [comment]
👍

**Comment by nickmshelley** [comment]
@hiltonc: https://blog.travis-ci.com/2013-08-21-abort-mission-cancel-running-builds/

I don't see that option. Is that because I'm not super owner guy like you?

**Comment by nickmshelley** [comment]
Never mind, the UI has changed. I guess it's an X now.

**Pull request merged**